src/src/Commands/MQTT.cpp

```diff
@@ -34,13 +42,13 @@ const __FlashStringHelper * Command_MQTT_Publish(struct EventStruct *event, cons
   const String topic = parseStringKeepCase(Line, 2);
   const String value = tolerantParseStringKeepCase(Line, 3);
   # ifndef BUILD_NO_DEBUG
-  addLog(LOG_LEVEL_DEBUG, concat(F("Publish: "), topic) + value);
-  #endif
+  addLog(LOG_LEVEL_DEBUG, strformat(F("Publish%s: %s:%s"), forceRetain ? F("R") : F(""), topic.c_str(), value.c_str()));
```

Comment:
`String(forceRetain ? F("R") : F("")).c_str()`
This F() is completely different on ESP8266, like this it may work on ESP32, but not on ESP8266.

Or you can use `forceRetain ? "R" : ""` as that's already a C-string and probably less memory usage compared to the 4-bytes per flash string and 16 bytes per `String()` cast.

Reply:
Fixed to use a `char`, it's merely a debug log, not activated that often.